pkg/webui/console/containers/device-onboarding-form/device-provisioning-form-section/index.js

```diff
@@ -38,36 +40,70 @@ import DeviceRegistrationFormSection, {
   initialValues as registrationInitialValues,
 } from './device-registration-form-section'
 
+const msg = defineMessages({
+  continue:
+    'To continue, please enter the JoinEUI of the end device so we can determine onboarding options',
+  confirm: 'Confirm',
+  reset: 'Reset',
+  registration: '✔ This end device can be registered on the network',
+  claiming: '✔ This end device can be claimed from its current owner',
+})
+
 const initialValues = merge(
   {
     ids: {
       join_eui: '',
     },
+    _claim: '',
   },
   claimingInitialValues,
   registrationInitialValues,
 )
 
-const DeviceTypeFormSection = () => {
-  const [isClaiming, setIsClaiming] = useState(undefined)
+const DeviceProvisioningFormSection = () => {
+  const dispatch = useDispatch()
+  const { values, setFieldValue, setValues } = useFormContext()
   const {
-    values: {
-      _inputMethod,
-      version_ids: version,
-      frequency_plan_id,
-      lorawan_version,
-      lorawan_phy_version,
-    },
-  } = useFormContext()
+    _claim,
+    _inputMethod,
+    version_ids: version,
+    frequency_plan_id,
+    lorawan_version,
+    lorawan_phy_version,
+    ids,
+  } = values
+
   const template = useSelector(selectDeviceTemplate)
   const mayProvisionDevice =
     (Boolean(frequency_plan_id) && Boolean(lorawan_version) && Boolean(lorawan_phy_version)) ||
     (_inputMethod === 'device-repository' && hasValidDeviceRepositoryType(version, template))
+  const mayConfirm = _claim === '' || _claim === undefined
```

Comment:
```suggestion
  const mayConfirm = ids.join_eui.length === 16
```
The button should be disabled when the Join EUI hasn't been fully entered. This was not precise in the wireframes. Sorry about that.